app/core/models.py

```diff
@@ -1,10 +1,73 @@
+import logging
+import json
 from django.db import models
 from model_utils.models import TimeStampedModel
 
+logger = logging.getLogger('dict_config_logger')
+
 
 class SchemaLedger(TimeStampedModel):
     """Model for Uploaded Schemas"""
+    SCHEMA_STATUS_CHOICES = [('published', 'published'),
+                             ('retired', 'retired')]
+
     schema_name = models.CharField(max_length=255)
     schema_iri = models.CharField(max_length=255)
-    schema_metadata = models.FileField(upload_to='schemas/')
-    status = models.CharField(max_length=50)
+    schema_file = models.FileField(upload_to='schemas/',
+                                   null=True,
+                                   blank=True)
+    status = models.CharField(max_length=10,
+                              choices=SCHEMA_STATUS_CHOICES)
+    metadata = models.JSONField(blank=True,
+                                help_text="auto populated from uploaded file")
+    version = models.CharField(max_length=6,
+                               help_text="auto populated from other version "
+                                         "fields")
+    major_version = models.SmallIntegerField()
+    minor_version = models.SmallIntegerField()
+    revision = models.SmallIntegerField()
+
+    def clean(self):
+        # store the contents of the file in the metadata field
+        if self.schema_file:
+            json_file = self.schema_file  
+            json_obj = json.load(json_file) # deserialises it
+
+            self.metadata = json_obj
+            json_file.close()
+            self.schema_file = None
+
+        # combine the versions
+        version = str(self.major_version) + '.' + str(self.minor_version) \
+            + '.' + str(self.revision)
+        self.version = version
+
+
+class TransformationLedger(TimeStampedModel):
+    """Model for Uploaded schema transformation mappings"""
```

Comment:
Not sure if it should be enforceable here, but what do we think about only supporting transforms for schemas that are stored within the system.  For example if you want to upload a transform for Schema A -> Schema B, then both schemas must already have records in the system.  This could be enforced via foreign key relationships, and in the UI it would be enforced with dropdown form selectors.

Reply:
I agree with this, I think it will give us some traceability for when schemas are removed/updated. I'll log a story for next sprint since it may take a bit more work - https://jira.deloitte.com/browse/ECC-817